kernel/kernel-5.10/drivers/usb/core/hub.c

```diff
@@ -5560,9 +5568,19 @@ static void port_event(struct usb_hub *hub, int port1)
 		dev_dbg(&port_dev->dev, "do warm reset\n");
 		if (!udev || !(portstatus & USB_PORT_STAT_CONNECTION)
 				|| udev->state == USB_STATE_NOTATTACHED) {
-			if (hub_port_reset(hub, port1, NULL,
-					HUB_BH_RESET_TIME, true) < 0)
-				hub_port_disable(hub, port1, 1);
+			/* Check for warm reset workaround */
+			if (!warm_reset_workaround) {
+				if (hub_port_reset(hub, port1, NULL,
+							HUB_BH_RESET_TIME, true) < 0)
+					hub_port_disable(hub, port1, 1);
+			} else {
+				if (strcmp(port_dev->dev.kobj.name, "2-3-port4") == 0) {
```

Comment:
Mind to use `kobject_name(&port_dev->dev.kobj)` and probably check on NULL?

Reply:
not at all, this part was ported from the Siemens BSP kernel (as pointed out in the commit message). I will improve it as you suggested....